qa/tasks/cephfs/test_cephfs_shell.py

```diff
@@ -943,50 +947,70 @@ def test_help(self):
-        self.fs.mon_manager.raw_cluster_cmd('config', 'set', 'client',
-                                            'debug_shell', str(not debugval))
-        output = self.get_cephfs_shell_cmd_output('set debug')
-        new_debug_val = \
-            str_to_bool(output[output.find('debug: ') + len('debug: ') : ])
-        assert not debugval == new_debug_val
-
-    def test_reading_conf_after_resetting_opt(self):
-        debugval = self.fs.mon_manager.raw_cluster_cmd('config', 'get',
-                                                       'client','debug_shell')
-        debugval = str_to_bool(debugval)
-
-        self.fs.mon_manager.raw_cluster_cmd('config', 'set', 'client',
-                                            'debug_shell', str(not debugval))
-        self.fs.mon_manager.raw_cluster_cmd('config', 'rm', 'client',
-                                            'debug_shell')
-        self.fs.mon_manager.raw_cluster_cmd('config', 'set', 'client',
-                                            'debug_shell', str(not debugval))
-        output = self.get_cephfs_shell_cmd_output('set debug')
-        new_debug_val = \
-            str_to_bool(output[output.find('debug: ') + len('debug: ') : ])
-        assert not debugval == new_debug_val
+        editor_val = self.get_cephfs_shell_cmd_output('set editor')
+        editor_val = editor_val.replace("'", "", 2)
+        editor_val = editor_val.split(': ')[1]
+        new_editor_val = '???'
+
+        tempconfpath = self.mount_a.client_remote.mktemp(
+            suffix='cephfs-shell.conf')
+        confcontents = "[cephfs-shell]\neditor = ???\neditor = %s\n" % \
+                       (editor_val)
+        sudo_write_file(self.mount_a.client_remote, tempconfpath, confcontents)
+
+        final_editor_val = self.get_cephfs_shell_cmd_output(
+            cmd='set editor', shell_conf_path=tempconfpath)
+        final_editor_val = final_editor_val.split(': ')[1]
+        final_editor_val = final_editor_val.replace("'", "", 2)
+
+        self.assertEqual(editor_val, final_editor_val)
+
+    def test_setting_opt_after_reading_conf(self):
+        editor_val = self.get_cephfs_shell_cmd_output('set editor')
+        editor_val = editor_val.replace("'", "", 2)
+        editor_val = editor_val.split(': ')[1]
+        new_editor_val = '???'
+
+        tempconfpath = self.mount_a.client_remote.mktemp(
+            suffix='cephfs-shell.conf')
+        confcontents = "[cephfs-shell]\neditor = " + new_editor_val
+        sudo_write_file(self.mount_a.client_remote, tempconfpath,confcontents)
+        # output of the command executed in next line must look as follows -
+        # $ cephfs-shell -- set editor vim, set editor
+        # editor - was: vim
+        # now: vim
+        # editor: vim
+        final_editor_val = self.get_cephfs_shell_cmd_output(cmd=('set editor '
+            '%s, set editor' % (editor_val)), shell_conf_path=tempconfpath)
```

Comment:
Also this code is repeated. Create a new method with cmd and confcontents as parameters?